### PR TITLE
Added ruff lint for jupyter notebook

### PR DESCRIPTION
**Pull request by Rjchauhan18**
# Description

- I have changed `.pre-commit-config.yaml` file and added  ruff instead of nbqa_ruff for lint the jupyter notebook.

Fixes # (issue)
- This PR fix this issue : https://github.com/pybamm-team/PyBaMM/issues/3476

## Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [ ] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -7,14 +7,20 @@ repos:
     rev: "v0.1.1"
     hooks:
       - id: ruff
-        args: [--fix, --ignore=E741, --exclude=__init__.py]
+        args: [--fix, --ignore=E741 , --exclude=__init__.py]
```
Comment (Saransh-cpp) [inline .pre-commit-config.yaml:10]:
```suggestion
        args: [--fix, --ignore=E741, --exclude=__init__.py]
        types_or: [python, pyi, jupyter]
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -7,14 +7,20 @@ repos:
...
-        args: ["--fix","--ignore=E501,E402"]
+      - id: ruff
+        types_or: [python, pyi, jupyter]
+
+  # - repo: https://github.com/nbQA-dev/nbQA
+  #   rev: 1.7.0
+  #   hooks:
+  #     - id: nbqa-ruff
+  #       additional_dependencies: [ruff==0.0.284]
+  #       args: ["--fix","--ignore=E501,E402"]
```
Comment (Saransh-cpp) [inline .pre-commit-config.yaml:23]: This can be deleted now
```suggestion
```

**Comment by Rjchauhan18** [comment]
Hello @Saransh-cpp ,
can you please review it?

**Tell me if any changes needed.**

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -7,14 +7,20 @@ repos:
...
-  - repo: https://github.com/nbQA-dev/nbQA
-    rev: 1.7.0
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.1.3
     hooks:
-      - id: nbqa-ruff
-        additional_dependencies: [ruff==0.0.284]
-        args: ["--fix","--ignore=E501,E402"]
+      - id: ruff
+        types_or: [python, pyi, jupyter]
```
Comment (Saransh-cpp) [inline .pre-commit-config.yaml:16]: We already have ruff-pre-commit in place, hence `type_or` should just be added to that section.

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Thanks, @Rjchauhan18! Some comments below -

File: .pre-commit-config.yaml (hunk 3 above)
Comment (Rjchauhan18) [reply .pre-commit-config.yaml:16]: okay , I will change it quickly.

Comment (Rjchauhan18) [reply .pre-commit-config.yaml]: Hello @Saransh-cpp 

Can you please review it again?

- I have removed nbqa_ruff's comments 
-  And fixes this too 
>We already have ruff-pre-commit in place, hence type_or should just be added to that section

**Comment by Rjchauhan18** [comment]
Hello @Saransh-cpp 

Actually , I never reverted any commit before and i search for it but it is confusing .

Can you telll me what exactly command i need to paste in terminal to revert the changes by pre-commit bot?

**Comment by Rjchauhan18** [comment]
Hey @Saransh-cpp 
I think I have reverted every commit commited by pre-commit bot .

**Comment by Rjchauhan18** [comment]
Oh ! no , it is again commiting the changes by itself.

**Comment by agriyakhetarpal** [comment]
You can test `ruff` and run `pre-commit` locally to test your changes (the easy way is to use `nox -s pre-commit` which runs all the hooks for you)

After that, you can drop specific commits made by `pre-commit.ci` on your branch by using an [interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History), and force push when you are done

**Comment by Rjchauhan18** [comment]
yes, I run ruff test locally , then commited only `.pre-commit-config.yaml` and drop another changes. but the problem is when  i commit then on github it  run  `.pre-commit-config.yaml` and   commit everthing on its own.

And  to run pre commit hooks i use the command as per guidelines in [contribution.md](https://github.com/pybamm-team/PyBaMM/blob/618b481f3c81d6519e0ed1b32a1b88c64eaaea8f/CONTRIBUTING.md?plain=1#L71)

**Comment by agriyakhetarpal** [comment]
If you fix/ignore the errors locally via configuring `ruff.toml` then `pre-commit.ci` should not commit changes by itself in theory

**Comment by Rjchauhan18** [comment]
so first i have to revert from my first commit

**Comment by Rjchauhan18** [comment]
Hey for now i need to delete my forked repositery of pybamm  and also clossing this pr for now i will make new fork and then make pr from there

**Comment by Saransh-cpp** [comment]
@Rjchauhan18 you don't need to delete your fork. You can push and revert commits in this branch itself. Actually, you should create a new branch and use that to create a PR. Pushing commits on develop (or the default branch) and creating a PR from that is not recommended.

**Pull request closed without merging**